tests/qos/test_buffer.py

```diff
@@ -0,0 +1,848 @@
+import logging
+import os
+import sys
+import time
+import re
+import json
+
+import pytest
+
+from tests.common import config_reload
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+
+profile_format = 'pg_lossless_{}_{}_profile'
+lossless_profile_pattern = 'pg_lossless_([1-9][0-9]*000)_([1-9][0-9]*m)_profile'
+
+default_cable_length_list = None
+default_lossless_headroom_data = None
+default_ingress_pool_number = 0
+default_mtu = '9100'
+
+testparam_headroom_override = None
+testparam_lossless_pg = None
+
+buffer_model_dynamic = True
+
+def detect_buffer_model(duthost):
+    global buffer_model_dynamic
+    buffer_model = duthost.shell('redis-cli -n 4 hget "DEVICE_METADATA|localhost" buffer_model')['stdout']
+    buffer_model_dynamic = buffer_model == 'dynamic'
```

Comment:
1. add () to buffer_model == 'dynamic' although not must
2. seems the function name and the function body are not aligned. does it check if the buffer module is dynamic? if so, change the function name to 'is_dynamic_buffer'
3. please add for each function simple header of description, expected values, return code.

Reply:
The function is to check the buffer model and save it. So to use `is_dynamic_buffer` is not accurate enough to me. I tend to keep the current name.
For functions called at setup phase of the module,
- functions starting with `detect` are responsible for detecting some fact of the system and storing them afterwards, like the buffer model, the number of ingress pools.
- functions starting with `load` are responsible for loading some data from database and storing them.

Will fix other comments